crates/core/tedge_mapper/src/mapper.rs

```diff
@@ -109,11 +116,13 @@ mod tests {
         let flock = check_another_instance_is_not_running(name)
             .expect("Another mapper instance is locking /run/lock/mapper_under_test.lock");
 
-        let mqtt_config = mqtt_client::Config::default().with_port(broker.port);
-        let mqtt_client = Client::connect(name, &mqtt_config).await?;
+        let mqtt_config = mqtt_channel::Config::default().with_port(broker.port);
+        let mqtt_client =
+            Connection::connect(name, &mqtt_config, TopicFilter::new_unchecked("in_topic")).await?;
```

Comment:
I would expect the tests no requiring a true connection anymore as the mapper only requires channels to be passed.

How about just creating an `UnboundedReceiver`, `UnboundedSender` channels and passing it to the mapper without expecting a connection and then just create msg in and expected out?

Reply:
It's indeed one of the purpose of this PR. However, I also tried to minimize code changes - notably for the tests. This refactoring introduce no features and I expect the tests to be unchanged.